File: packages/kotlin/src/main/kotlin/com/timelogic/direct/api/infrastructure/TransportConfiguration.kt
package com.timelogic.direct.api.infrastructure

object TransportConfiguration {
    const val DEFAULT_API_BASE_URL = "https://api.timelogicapi.com"
    const val DEFAULT_RAPID_API_HOST = "timelogic-api-world-time-timezones-time-calculations.p.rapidapi.com"
    const val DEFAULT_RAPID_API_BASE_URL = "https://$DEFAULT_RAPID_API_HOST"

    fun create(apiKey: String, rapidApi: Boolean = false, baseUrl: String? = null, rapidApiHost: String? = null): ApiClient {
        require(apiKey.isNotBlank()) { "apiKey is required" }
        val host = rapidApiHost?.takeIf { it.isNotBlank() } ?: DEFAULT_RAPID_API_HOST
        ApiClient.apiKey.clear()
        ApiClient.apiKeyPrefix.clear()
        ApiClient.accessToken = null
        val client = ApiClient(baseUrl?.trimEnd('/') ?: if (rapidApi) "https://$host" else DEFAULT_API_BASE_URL)
        if (rapidApi) {
            ApiClient.apiKey["X-RapidAPI-Key"] = apiKey.trim()
            ApiClient.apiKey["X-RapidAPI-Host"] = host
        } else {
            ApiClient.accessToken = apiKey.trim()
        }
        return client
    }
}

File: packages/csharp/src/TimeLogic.Api/Client/TransportConfiguration.cs
using System;

namespace TimeLogic.Api.Client
{
    /// <summary>Builds isolated direct or RapidAPI transport configurations.</summary>
    public static class TransportConfiguration
    {
        /// <summary>Default TimeLogic API base URL.</summary>
        public const string DefaultApiBaseUrl = "https://api.timelogicapi.com";
        /// <summary>Default RapidAPI host header value.</summary>
        public const string DefaultRapidApiHost = "timelogic-api-world-time-timezones-time-calculations.p.rapidapi.com";
        /// <summary>Default RapidAPI base URL.</summary>
        public const string DefaultRapidApiBaseUrl = "https://" + DefaultRapidApiHost;

        /// <summary>Creates an isolated configuration for the selected transport.</summary>
        public static Configuration Create(string apiKey, bool rapidApi = false, string baseUrl = null, string rapidApiHost = null)
        {
            if (string.IsNullOrWhiteSpace(apiKey)) throw new ArgumentException("apiKey is required", nameof(apiKey));
            var host = string.IsNullOrWhiteSpace(rapidApiHost) ? DefaultRapidApiHost : rapidApiHost.Trim();
            var configuration = new Configuration
            {
                BasePath = string.IsNullOrWhiteSpace(baseUrl) ? (rapidApi ? "https://" + host : DefaultApiBaseUrl) : baseUrl.TrimEnd('/')
            };
            if (rapidApi)
            {
                configuration.ApiKey["X-RapidAPI-Key"] = apiKey.Trim();
                configuration.ApiKey["X-RapidAPI-Host"] = host;
            }
            else
            {
                configuration.AccessToken = apiKey.Trim();
            }
            return configuration;
        }
    }
}
